src/octrees.rs

```diff
@@ -0,0 +1,147 @@
+use crate::bbox::BBox;
+use crate::buffers::{OutputBuffer};
+use crate::vector::Vec3;
+
+/// Pointer to a child node, which can be null.
+type Child<T> = Option<Box<T>>;
+
+/// Octree node
+pub struct OctNode {
+    /// 8 children. This is always either completely empty or completely full 
+    children: [Child<OctNode>; 8],
```

Comment:
Maybe Royce was right, try using

`Vec<OctNode>` instead of this to keep it simple.